scalableminds/webknossos
# Allow  to download NMLs of older versions for read-only tracings

Allow to display the version restore view for read-only tracings. Offer to download an NML of older versions.

@philippotto I also renamed the APITracingType and related types/variables to APIAnnotationType etc in this PR. For review I would suggest to exclude this commit to get a better sense of the actual changes :)

### URL of deployed dev instance (used for testing):
- https://readonlyversiondownload.webknossos.xyz

### Steps to test:
- Create a tracing with multiple versions and share it publicly. Copy the link. Login as another user and open the tracing. The tracing should be read-only.
  Preview an older version, the button should say "Download" instead of "Restore" and should download an NML of the previewed older version.
- Closing the version restore view should NOT make the read-only tracing modifiable.
- Opening/Closing the version view for modifiable tracings should work as before. Restoring should work as well.

### Issues:
- fixes #3627 

------
~~- [ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
~~- [ ] Updated [documentation](../blob/master/docs) if applicable~~
~~- [ ] Needs datastore update after deployment~~
